### ISSUE TITLE
 Poor inference accuracy when loading YOLOv8 model weights via YAML configuration

### ISSUE THREAD

**Issue by WindFreeFoliage** [ultralytics/ultralytics#18068]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I have trained a model using YOLOv8 and have generated the corresponding weights. When I initialize the model with 
model = YOLO('best.pt')

everything works as expected, and the predictions are accurate.

However, when I try to initialize the model using the YAML configuration file that I am certain matches the one used during training:

model = YOLO('train.yaml')
model.load('best.pt')

The inference accuracy becomes very poor, with noticeable deviations in the predictions.

Could someone please help me understand why this discrepancy occurs and provide guidance on how to resolve it?

Thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WindFreeFoliage, thank you for bringing this to our attention 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that we can use to investigate the issue further. This helps us debug more effectively. For now, please also ensure that your YAML configuration file exactly matches your training setup and does not unintentionally alter any settings.

If this is a ❓ Question about usage, please share any additional details such as the full YAML configuration, relevant snippets of your code, and specific examples or logs that highlight the accuracy discrepancies. It’s also worth confirming that you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To collaborate with the community or get real-time assistance, consider joining us on [Discord](https://discord.com/invite/ultralytics) 🎧. For more structured discussions, feel free to visit our [Discourse](https://community.ultralytics.com) or [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

First, ensure you are using the latest `ultralytics` package to rule out any resolved issues. Upgrade your environment with the command below:

```bash
pip install -U ultralytics
```

Verify all dependencies, including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/), are satisfied and that you are using the latest supported versions.

## Environments

To further troubleshoot, you can try testing your setup in the following verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Images** for consistent environments: See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response to help guide you. An Ultralytics engineer will also assist you as soon as possible. Stay tuned! 😊

**Comment by Y-T-G** [comment]
What model is this?

Are all the weights loaded when you use `.load()`?

**Comment by WindFreeFoliage** [comment]
> What model is this? Detect/Segment/Pose/OBB/Classify
> 
> Are all the weights loaded when you use `.load()`?

Hello, it is a Classify model. I first trained a model using the yolov8n-cls.yaml file and obtained the best.pt weights. When loading the model, it shows that all weights have been loaded successfully

**Comment by Y-T-G** [comment]
You should probably specify the `task` argument in YOLO()

**Comment by WindFreeFoliage** [comment]
> You should probably specify the `task` argument in YOLO()

Hello,

I am working with a classification model using YOLOv8 (yolov8n-cls.yaml). Below is my YAML configuration file used for training, which resulted in the best.pt weights:
nc: 6 # number of classes
scales: # model compound scaling constants
  n: [0.33, 0.25, 1024] # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
  s: [0.33, 0.50, 1024] # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
  m: [0.67, 0.75, 768] # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
  l: [1.00, 1.00, 512] # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
  x: [1.00, 1.25, 512] # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs
backbone:
  - [-1, 1, Conv, [64, 3, 2]]
  - [-1, 1, Conv, [128, 3, 2]]
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]

head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]]
  - [-1, 3, C2f, [512]]
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]]
  - [-1, 3, C2f, [256]]
  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]
  - [-1, 3, C2f, [512]]
  - [-1, 1, Classify, [nc]] # Classify

When I initialize the model directly with model = YOLO('best.pt',task='classify'), it works as expected, and the predictions are accurate.

However, when I initialize the model using the YAML configuration and then load the weights:
model = YOLO('model.yaml',task='classify')
model.load('best.pt')

the predictions do not display the class labels but instead output numeric values. the previously mentioned accuracy issues still persist

Could someone please help me understand why this discrepancy occurs and provide guidance on how to resolve it? Specifically, I would like to know why loading the model through the YAML configuration changes the prediction output from class labels to numeric values, and why it leads to a significant drop in accuracy.

**Comment by WindFreeFoliage** [comment]
Additionally, even though all weights are reported as loaded successfully (Transferred 248/248 items from pretrained weights)

**Comment by glenn-jocher** [comment]
The issue may stem from discrepancies between the model architecture defined in your YAML file and the one saved in the `best.pt` weights. To resolve this, ensure the `best.pt` file is directly initialized, as it contains the precise architecture and parameters used during training. Using `model = YOLO('

**Comment by Y-T-G** [comment]
Try upgrading to the latest version.

It might be fixed by https://github.com/ultralytics/ultralytics/pull/18085
